os/net/netdev/netdev_dhcpd.c

```diff
@@ -0,0 +1,165 @@
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>		/* TinyAra configuration */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <unistd.h>
+#include <time.h>
+#include <errno.h>
+#include <debug.h>				/* For ndbg, vdbg */
+
+#include <net/if.h>
+#include <netinet/in.h>
+#include <semaphore.h>
+#include <pthread.h>
+#include <netutils/netlib.h>
+#include <protocols/dhcpd.h>	/* Advertised DHCPD APIs */
+
+#include <net/lwip/netif.h>
+#include <net/lwip/netifapi.h>
+
+/****************************************************************************
+ * Global Data
+ ****************************************************************************/
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+#define DHCP_SERVER_PORT         67
+#define DHCP_CLIENT_PORT         68
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Global Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: dhcps_server_status
+ ****************************************************************************/
+int netdev_dhcp_server_status(char *intf)
```

Comment:
You using like a below
```
		ret = netdev_dhcp_server_status((char *)in_arg->host_name);
		if (ret != 0) {
			printf("stop dhcpd fail\n");
		}
```
If server is opened, it returns '1' is it correct????

Reply:
This is wrong. It should return 0 when DHCP is opend. Thanks.